### PR TITLE
MSSQL support for Inbox

### PR DESCRIPTION
**Pull request by fenek**
This PR adds MSSQL support in `mod_inbox_odbc`.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[5054.1](https://travis-ci.org/esl/MongooseIM/jobs/400682574) / Erlang 19.3 / odbc_mssql_mnesia / fc1abccf3b56a752e25d00baf1a6fbc525b8a70a
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1965/5054/odbc_mssql_mnesia.19.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5054/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-01a5aba6-e339-48cc-9152-7f245bc56a3a.2018-07-06_01.09.33/index.html)
**OK: 4** / **Failed: 6** / User-skipped: 0 / **Auto-skipped: 48** 
<details><summary>inbox_SUITE:one_to_one:msg_sent_stored_in_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.838.0>,
          [{event_manager,<0.837.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.837.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"AE40C481046E21ED">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-msg_sent_stored_in_inbox/1-fun-2-',2,
       [{file,"inbox_SUITE.erl"},{line,255}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1,6,
       [{fi...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5054/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-01a5aba6-e339-48cc-9152-7f245bc56a3a.2018-07-06_01.09.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.09.36/inbox_suite.msg_sent_stored_in_inbox.html)
</details>
<details><summary>inbox_SUITE:one_to_one:msg_sent_stored_in_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.909.0>,
          [{event_manager,<0.908.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.908.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"E2ABC9BC58C232C2">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-msg_sent_stored_in_inbox/1-fun-2-',2,
       [{file,"inbox_SUITE.erl"},{line,255}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1,6,
       [{fi...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5054/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-01a5aba6-e339-48cc-9152-7f245bc56a3a.2018-07-06_01.09.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.09.36/inbox_suite.msg_sent_stored_in_inbox.22.html)
</details>
<details><summary>inbox_SUITE:one_to_one:msg_sent_stored_in_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.978.0>,
          [{event_manager,<0.977.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.977.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"A2803D2A553E1569">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-msg_sent_stored_in_inbox/1-fun-2-',2,
       [{file,"inbox_SUITE.erl"},{line,255}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1,6,
       [{fi...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5054/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-01a5aba6-e339-48cc-9152-7f245bc56a3a.2018-07-06_01.09.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.09.36/inbox_suite.msg_sent_stored_in_inbox.38.html)
</details>
<details><summary>inbox_SUITE:muclight:simple_groupchat_stored_in_all_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.1025.0>,
          [{event_manager,<0.1024.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.1024.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"37476D886C329F21">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-simple_groupchat_stored_in_all_inbox/1-fun-3-',3,
       [{file,"inbox_SUITE.erl"},{line,430}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5054/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-01a5aba6-e339-48cc-9152-7f245bc56a3a.2018-07-06_01.09.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.09.36/inbox_suite.simple_groupchat_stored_in_all_inbox.html)
</details>
<details><summary>inbox_SUITE:muclight:simple_groupchat_stored_in_all_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.1078.0>,
          [{event_manager,<0.1077.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.1077.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"EBC877DBF75A54EF">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-simple_groupchat_stored_in_all_inbox/1-fun-3-',3,
       [{file,"inbox_SUITE.erl"},{line,430}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5054/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-01a5aba6-e339-48cc-9152-7f245bc56a3a.2018-07-06_01.09.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.09.36/inbox_suite.simple_groupchat_stored_in_all_inbox.58.html)
</details>
<details><summary>inbox_SUITE:muclight:simple_groupchat_stored_in_all_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.1131.0>,
          [{event_manager,<0.1130.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.1130.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"2C765D94CDA02070">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-simple_groupchat_stored_in_all_inbox/1-fun-3-',3,
       [{file,"inbox_SUITE.erl"},{line,430}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5054/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-01a5aba6-e339-48cc-9152-7f245bc56a3a.2018-07-06_01.09.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.09.36/inbox_suite.simple_groupchat_stored_in_all_inbox.70.html)
</details>

**Comment by mongoose-im** [comment]
[5055.1](https://travis-ci.org/esl/MongooseIM/jobs/400689592) / Erlang 19.3 / odbc_mssql_mnesia / b7cc8ebc422f7a3a3c4dde97413898e8d13caeee
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1965/5055/odbc_mssql_mnesia.19.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5055/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-85ec9158-f41a-471b-af07-df94ddc60f32.2018-07-06_01.39.33/index.html)
**OK: 4** / **Failed: 6** / User-skipped: 0 / **Auto-skipped: 48** 
<details><summary>inbox_SUITE:one_to_one:msg_sent_stored_in_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.838.0>,
          [{event_manager,<0.837.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.837.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"64CA3A4A80AF8859">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-msg_sent_stored_in_inbox/1-fun-2-',2,
       [{file,"inbox_SUITE.erl"},{line,255}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1,6,
       [{fi...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5055/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-85ec9158-f41a-471b-af07-df94ddc60f32.2018-07-06_01.39.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.39.36/inbox_suite.msg_sent_stored_in_inbox.html)
</details>
<details><summary>inbox_SUITE:one_to_one:msg_sent_stored_in_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.909.0>,
          [{event_manager,<0.908.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.908.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"AEDF321B444AAA6F">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-msg_sent_stored_in_inbox/1-fun-2-',2,
       [{file,"inbox_SUITE.erl"},{line,255}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1,6,
       [{fi...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5055/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-85ec9158-f41a-471b-af07-df94ddc60f32.2018-07-06_01.39.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.39.36/inbox_suite.msg_sent_stored_in_inbox.18.html)
</details>
<details><summary>inbox_SUITE:one_to_one:msg_sent_stored_in_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.978.0>,
          [{event_manager,<0.977.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.977.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"1DB01048F200B5AC">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-msg_sent_stored_in_inbox/1-fun-2-',2,
       [{file,"inbox_SUITE.erl"},{line,255}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1,6,
       [{fi...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5055/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-85ec9158-f41a-471b-af07-df94ddc60f32.2018-07-06_01.39.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.39.36/inbox_suite.msg_sent_stored_in_inbox.34.html)
</details>
<details><summary>inbox_SUITE:muclight:simple_groupchat_stored_in_all_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.1025.0>,
          [{event_manager,<0.1024.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.1024.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"3755AF082D0EED89">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-simple_groupchat_stored_in_all_inbox/1-fun-3-',3,
       [{file,"inbox_SUITE.erl"},{line,430}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5055/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-85ec9158-f41a-471b-af07-df94ddc60f32.2018-07-06_01.39.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.39.36/inbox_suite.simple_groupchat_stored_in_all_inbox.html)
</details>
<details><summary>inbox_SUITE:muclight:simple_groupchat_stored_in_all_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.1078.0>,
          [{event_manager,<0.1077.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.1077.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"59575646FB25A8DC">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-simple_groupchat_stored_in_all_inbox/1-fun-3-',3,
       [{file,"inbox_SUITE.erl"},{line,430}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5055/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-85ec9158-f41a-471b-af07-df94ddc60f32.2018-07-06_01.39.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.39.36/inbox_suite.simple_groupchat_stored_in_all_inbox.54.html)
</details>
<details><summary>inbox_SUITE:muclight:simple_groupchat_stored_in_all_inbox</summary>


```erlang
{error,
  {timeout_when_waiting_for_stanza,
    [{escalus_client,wait_for_stanza,
       [{client,<<"alicE@localhost/res1">>,escalus_tcp,<0.1131.0>,
          [{event_manager,<0.1130.0>},
           {server,<<"localhost">>},
           {username,<<"alicE">>},
           {resource,<<"res1">>}],
          [{event_client,
             [{event_manager,<0.1130.0>},
            {server,<<"localhost">>},
            {username,<<"alicE">>},
            {resource,<<"res1">>}]},
           {resource,<<"res1">>},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {host,<<"localhost">>},
           {port,5222},
           {auth,{escalus_auth,auth_plain}},
           {wspath,undefined},
           {username,<<"alicE">>},
           {server,<<"localhost">>},
           {password,<<"matygrysa">>},
           {stream_id,<<"BED6C0B3F2F2E7E5">>}]},
        5000],
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_client.erl"},
        {line,138}]},
     {inbox_SUITE,check_inbox,3,[{file,"inbox_SUITE.erl"},{line,892}]},
     {inbox_SUITE,'-simple_groupchat_stored_in_all_inbox/1-fun-3-',3,
       [{file,"inbox_SUITE.erl"},{line,430}]},
     {escalus_story,story,4,
       [{file,
          "/home/travis/build/esl/MongooseIM/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
     {test_server,run_test_case_eval1...
```

[Report log](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5055/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-85ec9158-f41a-471b-af07-df94ddc60f32.2018-07-06_01.39.33/big_tests.tests.inbox_SUITE.logs/run.2018-07-06_01.39.36/inbox_suite.simple_groupchat_stored_in_all_inbox.66.html)
</details>

**Comment by mongoose-im** [comment]
[5058.1](https://travis-ci.org/esl/MongooseIM/jobs/400707471) / Erlang 19.3 / odbc_mssql_mnesia / eb1a52c679a231aa5ef0bf5f6914fc614a06233a
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1965/5058/odbc_mssql_mnesia.19.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5058/odbc_mssql_mnesia.19.3/big/ct_run.test%40travis-job-0aad2e25-97ba-4fc0-b5e3-002ffd36c8a8.2018-07-06_03.06.23/index.html)
**OK: 20** / Failed: 0 / User-skipped: 0 / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[5059.1](https://travis-ci.org/esl/MongooseIM/jobs/400709530) / Erlang 19.3 / small_tests / 7feed4b42bf370b846dfa8b133c9ec7872c5921f
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1965/5059/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5059/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-07-06_03.13.30/lib.mongooseim.logs/run.2018-07-06_03.13.31/suite.log.html)

**Comment by mongoose-im** [comment]
[5060.1](https://travis-ci.org/esl/MongooseIM/jobs/400779561) / Erlang 19.3 / small_tests / c2ad4d8f5d46cf5d71925df3166e18c529b27f0d
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1965/5060/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1965/5060/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-07-06_08.22.27/lib.mongooseim.logs/run.2018-07-06_08.22.28/suite.log.html)

File: src/inbox/mod_inbox_odbc_mysql.erl [hunk 1]
```diff
@@ -21,11 +27,15 @@
                 Count :: binary(),
                 MsgId :: binary()) -> query_result().
 set_inbox(Username, Server, ToBareJid, Content, Count, MsgId) ->
-    mongoose_rdbms:sql_query(Server, ["insert into inbox(luser, lserver, remote_bare_jid, content,
-    unread_count, msg_id) values (", ?ESC(Username), ",", ?ESC(Server), ",", ?ESC(ToBareJid), ",",
-        ?ESC(Content), ",", ?ESC(Count), ",", ?ESC(MsgId),
-        ") on duplicate key update content=",
-        ?ESC(Content), ", unread_count=", ?ESC(Count), ",msg_id=", ?ESC(MsgId), ";"]).
+    mongoose_rdbms:sql_query(Server,
+        ["insert into inbox(luser, lserver, remote_bare_jid, content, unread_count, msg_id) "
```
Comment (michalwski) [inline src/inbox/mod_inbox_odbc_mysql.erl]: 👍 that's much more readable!

File: doc/modules/mod_inbox.md [hunk 2]
```diff
@@ -19,6 +19,15 @@ Only changes that affect the user directly will be stored in their inbox.
 If true, the inbox conversation is removed for a user when they are removed from the groupchat.
 * **iqdisc** (atom, default: `no_queue`)
 
+### Note about supported RDBMS
+
+`mod_inbox` executes upsert queries, which have different syntax in every supported RDBMS.
+Inbox currently supports following DBs:
```
Comment (goddammit) [inline doc/modules/mod_inbox.md]: Inbox currently supports the following DBs:

**Pull request merged**